src/transformers/models/gemma2/modular_gemma2.py

```diff
@@ -290,9 +290,12 @@ def tanh_softcap(score, b, h, q_idx, kv_idx):
         return_lse=output_attentions,
     )
     if not output_attentions:
-        return attn_output, None
+        attn_weights = None
     else:
-        return attn_output[0], attn_output[1]
+        attn_output, attn_weights = attn_output
+
+    attn_output = attn_output.transpose(1, 2).contiguous()
+    return attn_output[0], attn_weights
```

Comment:
```suggestion
    return attn_output, attn_weights
```
Otherwise we only take the first batch element if i see it correctly 👀

Reply:
Indeed, that's a typo!